sormas-backend/src/main/java/de/symeda/sormas/backend/person/PersonFacadeEjb.java

```diff
@@ -1859,6 +1866,84 @@ public void mergePerson(String leadPersonUuid, String otherPersonUuid, boolean m
 		service.ensurePersisted(leadPerson);
 	}
 
+	private void processPersonAddressMerge(PersonDto leadPersonDto, PersonDto otherPersonDto) {
+		if (differentLocation(leadPersonDto.getAddress(), otherPersonDto.getAddress())) {
+			LocationDto otherAddress = otherPersonDto.getAddress();
+			otherAddress.setAddressType(PersonAddressType.OTHER_ADDRESS);
+			otherAddress.setAddressTypeDetails(I18nProperties.getString(Strings.messagePersonMergedAddressDescription));
+			leadPersonDto.addAddress(otherAddress);
+		} else {
+			DtoHelper.copyDtoValues(leadPersonDto.getAddress(), otherPersonDto.getAddress(), false);
+		}
+	}
+
+	private boolean differentLocation(LocationDto firstAddress, LocationDto secondAddress) {
```

Comment:
```suggestion
	private boolean areDifferentLocations(LocationDto firstAddress, LocationDto secondAddress) {
```

Also, I'd move this method to LocationFacadeEjb